src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if ((node.key == key || (key != null && key.equals(node.key)))) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if ((node.key == key || (key != null && key.equals(node.key)))) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int indexFor(int hash, int size) {
```

Comment:
You can actually create only one method instead of indexOf and hash
```
private int calculateIndex(K key) {
     return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
}
```